src/Commands/Connection/InfoCommand.php

```diff
@@ -0,0 +1,84 @@
+<?php
+
+namespace Pantheon\Terminus\Commands\Connection;
+
+use Consolidation\OutputFormatters\StructuredData\RowsOfFields;
+
+use Pantheon\Terminus\Commands\TerminusCommand;
+use Terminus\Collections\Sites;
+use Terminus\Models\Environment;
+
+/**
+ * Class InfoCommand
+ * @package Pantheon\Terminus\Commands\Connection
+ */
+class InfoCommand extends TerminusCommand
+{
+    /**
+     * Retrieve connection info for a specific environment such as git, sftp, mysql, redis
+     *
+     * @authorized
+     *
+     * @command connection:info
+     *
+     * @param string $environment Name of the environment to retrieve
+     * @param string $filter Parameter filter (optional)
+     * @param array $options [fields=<env,param,value>] [format=<table|csv|yaml|json>]
+     *
+     * @return RowsOfFields
+     *
+     * @field-labels
+     *   env: Environment
+     *   param: Parameter
+     *   value: Connection Info
+     *
+     * @example connection:info awesome-site dev git_command --format=json
+     *   Display connection information only for the given parameter
+     *
+     */
+    public function connectionInfo(
+        $environment,
+        $filter = null,
+        $options = ['format' => 'table', 'fields' => 'param,value']
+    ) {
+        $site_env = explode('.', $environment);
```

Comment:
Input args can be adjusted in an interact hook.  We can define an interact hook that is attached to any command with a given annotation, so we could have a `@site-arg` annotation and a `@site-env-arg` annotation, so each command that took a site, or a site and an env, could declare which arguments it needs.  The hook could also consult the .env file and etc. as needed to come up with the right answer.  If the site and env could not be satisfactorily determined, then the hook could prompt the user and ask them which site and env they meant.